packages/agent/src/oidc.ts

```diff
@@ -600,18 +594,116 @@ function encryptAuthResponse({
   return compactJwe;
 }
 
+export async function createPermissionGrants(
+  selectedDids: string[],
+  ephemeralDid: BearerDid,
+  dwn: AgentDwnApi
+) {
+  const grantPromises = selectedDids.map(async (did) => {
+    // TODO: Replace with real permission request
+    const permissionRequestData = {
+      description : 'The app is asking to Records Write to http://profile-protocol.xyz',
+      scope       : {
+        interface : DwnInterfaceName.Records,
+        method    : DwnMethodName.Write,
+        protocol  : 'http://profile-protocol.xyz',
+      },
+    };
+
+    // TODO: Confirm this
+    const message = await dwn.processRequest({
+      author        : did,
+      target        : did,
+      messageType   : DwnInterface.RecordsWrite,
+      messageParams : {
+        recipient    : ephemeralDid.uri,
+        protocolPath : 'grant',
+        protocol     : ' https://tbd.website/dwn/permissions',
+        dataFormat   : 'application/json',
+        data         : Convert.object(permissionRequestData).toUint8Array(),
```

Comment:
At a glance the data is missing `delegated: true`.

Reply:
as with the last comment useful info but i think it needs to be part of a later PR rather than this one